main.rb

```diff
@@ -0,0 +1,232 @@
+    length.times do
+      yield (self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr_ret = []
+    i = 0
+    length.times do
+      arr_ret.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    arr_ret
+  end
+
+  def my_all?(par = nil)
+    ret_value = true
+
+    if block_given?
+      i = 0
+      length.times do
+        return false unless yield(self[i])
+
+        i += 1
+      end
+    elsif !par
+      i = 0
+      length.times do
+        return false unless self[i]
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = false unless x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = false unless x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = false unless x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_any?(par = nil)
+    ret_value = false
+    if block_given?
+      i = 0
+      length.times do
+        return true if yield(self[i])
+
+        i += 1
+      end
+    elsif !par
+      i = 0
+      length.times do
+        return true if self[i]
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = true if x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = true if x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = true if x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_none?(par = nil)
+    ret_value = true
+    if block_given?
+      i = 0
+      length.times do
+        return false if yield(self[i])
+
+        i += 1
+      end
+    elsif !par
+      i = 0
+      length.times do
+        return false if self[i]
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = false if x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = false if x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = false if x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_count(par = nil)
+    counter = 0
+    if block_given?
+      i = 0
+      length.times do
+        counter += 1 unless yield(self[i]).nil?
+        i += 1
+      end
+    elsif !par
+      return length
+    else
+      if par.is_a? Regexp
+        my_each { |x| counter += 1 if x =~ par }
+        return counter
+      end
+      if par.is_a? Class
+        my_each { |x| counter += 1 if x.is_a?(par) }
+        return counter
+      end
+      unless par.nil?
+        my_each { |x| counter += 1 if x == par }
+      end
+    end
+    counter
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] my_count counts the number of elements yielding a true value if a block is given
```ruby
[5, 4, 1].my_count { |x| x > 2 } == [5, 4, 1].count { |x| x > 2 } => 2
```
- Yours is giving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/77704490-cc2aa580-6f82-11ea-814b-7f17ed46c862.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77704518-d8166780-6f82-11ea-8265-13823bf6e785.png)